cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -253,4 +258,23 @@ public void releaseCallResources() {
 
     return evaluated;
   }
+
+  /**
+   * Get preferences for the supplied namespace.
+   *
+   * @param namespace the name of the namespace to fetch preferences for.
+   * @param resolved true if resolved properties are desired.
+   * @return Map containing the preferences for this namespace
+   * @throws IOException if the preferencesFetcher could not complete the request.
+   * @throws IllegalArgumentException if the supplied namespace doesn't exist.
+   */
+  @Override
+  public Map<String, String> getPreferencesForNamespace(String namespace, boolean resolved)
+    throws IOException, IllegalArgumentException {
+    try {
+      return preferencesFetcher.get(new NamespaceId(namespace), resolved).getProperties();
+    } catch (NotFoundException nfe) {
+      throw new IllegalArgumentException("Invalid namespace: " + namespace, nfe);
```

Comment:
It's better to have the message as `"Namespace '" + namespace + "' does not exist"`

Reply:
I've updated the error message.